apps/base-docs/tutorials/docs/1_thirdweb-unreal-nft-items.md

```diff
@@ -0,0 +1,773 @@
+```
+
+Then, add the function to `ThirdwebManager.cpp`. It's similar, but instead hits the endpoint for the NFT color array and uses the response you just created. It also expects the response to be an array of strings instead of searching for a property called `result`:
+
+```c++
+// ThirdwebManager.cpp
+void AThirdwebManager::GetNFTColors()
+{
+	TSharedRef<IHttpRequest, ESPMode::ThreadSafe> HttpRequest = FHttpModule::Get().CreateRequest();
+	HttpRequest->SetURL(this->ServerUrl + "/engine/get-nft-colors"); // The endpoint to get the NFT colors
+	HttpRequest->SetVerb("POST");
+	HttpRequest->SetHeader(TEXT("Content-Type"), TEXT("application/json"));
+
+	TSharedPtr<FJsonObject> JsonObject = MakeShareable(new FJsonObject);
+	JsonObject->SetStringField("authToken", AuthToken);
+
+	FString OutputString;
+	TSharedRef<TJsonWriter<>> Writer = TJsonWriterFactory<>::Create(&OutputString);
+	FJsonSerializer::Serialize(JsonObject.ToSharedRef(), Writer);
+
+	UE_LOG(LogTemp, Warning, TEXT("OutputString: %s"), *OutputString);
+
+	HttpRequest->SetContentAsString(OutputString);
+
+	HttpRequest->OnProcessRequestComplete().BindLambda([this](FHttpRequestPtr Request, FHttpResponsePtr Response, bool bWasSuccessful)
+		{
+		if (bWasSuccessful && Response.IsValid())
+		{
+			int32 StatusCode = Response->GetResponseCode();
+			if (StatusCode == 200)
+			{
+				TArray<TSharedPtr<FJsonValue>> JsonArray;
+				TSharedRef<TJsonReader<>> Reader = TJsonReaderFactory<>::Create(Response->GetContentAsString());
+				if (FJsonSerializer::Deserialize(Reader, JsonArray) && JsonArray.Num() > 0)
+				{
+					TArray<FString> ResponseArray;
+					for (const TSharedPtr<FJsonValue>& Value : JsonArray)
+					{
+						FString StringValue;
+						if (Value->TryGetString(StringValue))
+						{
+							ResponseArray.Add(StringValue);
+						}
+					}
+					this->OnNFTColorsResponse.Broadcast(true, ResponseArray);
+					UE_LOG(LogTemp, Warning, TEXT("Get NFT Color response: %s"), *Response->GetContentAsString());
+					return;
+				}
+				this->OnNFTColorsResponse.Broadcast(false, TArray<FString>());
+			}
+			else
+			{
+				FString ErrorMsg = FString::Printf(TEXT("HTTP Error: %d, Response: %s"), StatusCode, *(Response->GetContentAsString()));
+				TArray<FString> ErrorArray;
+				ErrorArray.Add(ErrorMsg);
+				this->OnNFTColorsResponse.Broadcast(false, ErrorArray);
+				UE_LOG(LogTemp, Warning, TEXT("ErrorMsg: %s"), *ErrorMsg);
+			}
+		}
+		else
+		{
+			TArray<FString> ErrorArray;
+			ErrorArray.Add(TEXT("Failed to connect to the server."));
+			this->OnNFTColorsResponse.Broadcast(false, ErrorArray);
+			UE_LOG(LogTemp, Warning, TEXT("Failed to connect to the server."));
+		} });
+
+	HttpRequest->ProcessRequest();
+}
+```
+
+Finally, expose this function to the editor.
+
+```c++
+// ThirdwebManager.h
+// Function to perform the get NFT Colors operation
+UFUNCTION(BlueprintCallable, Category = "Thirdweb")
+void GetNFTColors();
+```
+
+Compile and reload the project in the editor.
```

Comment:
I'm not sure why github is rendering this as blue, but it looks normal in preview in code and running locally